dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -4780,6 +4733,29 @@ private Contentlet internalCheckin(Contentlet contentlet,
                 changedURI = ! oldURI.equals(identifier.getURI());
             }
 
+            //Relate the tags with the saved contentlet
+            for ( Entry<String, String> tagEntry : tagsValues.entrySet() ) {
+                //From the given CSV tags names list search for the tag objects and if does not exist create them
+                List<Tag> list = tagAPI.getTagsInText(tagEntry.getValue(), tagsHost);
+
+                // empty string for tag field value wipes out existing tags
+                if(UtilMethods.isSet(list) || tagEntry.getValue().equals("")) {
```

Comment:
probably would be safer to do 

StringPool.BLANK.equals(tagEntry.getValue())

Reply:
Done